assignment/core/src/main/java/com/assignment/core/crypto/RSACipherProvider.java

```diff
@@ -0,0 +1,57 @@
+package com.assignment.core.crypto;
+
+import org.osgi.service.component.annotations.Component;
+import javax.crypto.Cipher;
+import java.security.*;
+import java.util.Base64;
+
+@Component(
+        service = AsymmetricCipherProvider.class,
+        property = {
+                "algorithm=RSA"
+        }
+)
+public class RSACipherProvider implements AsymmetricCipherProvider {
+    private static final String ALGORITHM = "RSA";
+    private KeyPair keyPair;
+
+    public RSACipherProvider() {
+        try {
+            KeyPairGenerator keyGen = KeyPairGenerator.getInstance(ALGORITHM);
+            keyGen.initialize(2048);
+            this.keyPair = keyGen.generateKeyPair();
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Failed to initialize RSA key pair", e);
+        }
+    }
+
+    @Override
+    public String getAlgorithm() {
+        return ALGORITHM;
+    }
+
+    @Override
+    public String encrypt(String plaintext) {
+        try {
+            Cipher cipher = Cipher.getInstance(ALGORITHM);
+            cipher.init(Cipher.ENCRYPT_MODE, keyPair.getPublic());
+            byte[] encryptedBytes = cipher.doFinal(plaintext.getBytes());
+            return Base64.getEncoder().encodeToString(encryptedBytes);
+        } catch (Exception e) {
+            throw new RuntimeException("Encryption failed", e);
+        }
+    }
+
+    @Override
+    public String decrypt(String ciphertext) {
+        try {
+            Cipher cipher = Cipher.getInstance(ALGORITHM);
+            cipher.init(Cipher.DECRYPT_MODE, keyPair.getPrivate());
+            byte[] decryptedBytes = cipher.doFinal(Base64.getDecoder().decode(ciphertext));
+            return new String(decryptedBytes);
+        } catch (Exception e) {
+            throw new RuntimeException("Decryption failed", e);
```

Comment:
Does this seem like an appropriate use case to throw a `RuntimeException`

Reply:
removed it, please review changes